File: src/test-profiles/go-playwright/templates/example_test.go
// Example test — a runnable starting point you EDIT IN PLACE.
//
// Scaffolded by `saifctl init`, `saifctl init tests`, or `saifctl feat
// design-tests`. Unlike helpers.go / infra_test.go, this file is meant to
// be edited: keep it as a working reference for the playwright-go format,
// or replace its body with your real assertion. The scaffold skips this
// file when it already exists; pass `--force` to overwrite.
//
// What this demonstrates:
//   - Importing the shared transport from the helpers package
//   - Hitting the staging app via HTTPRequest helper
//   - Asserting on response status
//
// If your project doesn't expose an HTTP service in the staging container,
// delete this file and use ExecSidecar from helpers.go instead (CLI-style
// checks, like the go-gotest example).
package example_test

import (
	"testing"

	helpers "github.com/placeholder/factory-tests/helpers"
)

func TestExampleStagingAppRespondsAtRoot(t *testing.T) {
	resp, err := helpers.HTTPRequest("GET", "/", nil, nil)
	if err != nil {
		t.Fatalf("staging app not reachable: %v", err)
	}
	defer resp.Body.Close()

	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
		t.Errorf("expected 2xx/3xx, got %d", resp.StatusCode)
	}
}

File: src/test-profiles/rust-playwright/templates/example_test.rs
// Example test — a runnable starting point you EDIT IN PLACE.
//
// Scaffolded by `saifctl init`, `saifctl init tests`, or `saifctl feat
// design-tests`. Unlike helpers.rs / infra_test.rs, this file is meant to
// be edited: keep it as a working reference for the rust-playwright format,
// or replace its body with your real assertion. The scaffold skips this
// file when it already exists; pass `--force` to overwrite.
//
// What this demonstrates:
//   - Launching a Playwright browser via `new_page()` (async)
//   - Navigating to `base_url()` (the staging app)
//   - Asserting on response status
//
// If your project doesn't expose an HTTP service in the staging container,
// delete this file and use `exec_sidecar` from helpers.rs instead (CLI-style
// checks, like the rust-rusttest example).

use crate::helpers::{base_url, new_page};

#[tokio::test]
async fn example_staging_app_responds_at_root() {
    let (_pw, browser, page) = new_page().await.expect("failed to open browser page");
    let response = page
        .goto_builder(&base_url())
        .goto()
        .await
        .expect("page.goto failed");
    let status = response
        .status()
        .expect("could not read response status");
    assert!(
        (200..400).contains(&status),
        "expected 2xx/3xx, got {}",
        status,
    );
    browser.close().await.ok();
}
